src/main/java/com/oneops/api/resource/Transition.java

```
@@ -245,7 +232,142 @@ public Release commitEnvironment(String environmentName, List<Long> excludePlatf
+	 * 	 	b. If there are no errors in `comments` of the environment, throw `OneOpsClientAPIException`
+	 * 	  	with DEFAULT_ERROR_MESSAGE.
+	 *
+	 * @param environmentName
+	 * @return
+	 * @throws OneOpsClientAPIException
+	 */
+	public Deployment deploy(String environmentName, List<Long> excludePlatforms, List<Long> includeComponents, List<Long> includeClouds,
+							 boolean ignoreCloudDeploymentOrder, String comments) throws OneOpsClientAPIException {
+
+		RequestSpecification request = createRequest();
+		Map<String, String> properties = new HashMap<>();
+
+		addStringToPropertyMap(excludePlatforms, properties, "exclude_platforms");
+		addStringToPropertyMap(includeComponents, properties, "components");
+		addStringToPropertyMap(includeClouds, properties, "include_clouds");
+
+		properties.put("ignore_cloud_dpmt_order", String.valueOf(ignoreCloudDeploymentOrder));
+
+		ResourceObject ro = new ResourceObject();
+		ro.setProperties(properties);
+
+		JSONObject jsonObject = JsonUtil.createJsonObject(ro, null);
+
+		if (comments == null)
+			comments = String.format("Deployment for %s environment identified by: %s",
+					environmentName, UUID.randomUUID().toString());
+
+		Map<String, String> cmsDeployment = new HashMap<>();
+		cmsDeployment.put("comments", comments);
+		jsonObject.put("cms_deployment", cmsDeployment);
+
+		// Start a new deployment
+		CiResource envInfo = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+
+		while (!IConstants.DEFAULT_CI_STATE.equalsIgnoreCase(envInfo.getCiState())) {
+			LOG.info("Deployment initiation in progress ...");
+			try {
+				Thread.sleep(5000);
+			} catch(InterruptedException ie) {
+				LOG.error("Error waiting for deployment to start: {}", ie.getMessage());
+			}
+			envInfo = getEnvironment(environmentName);
+		}
+
+		Deployment latestDeployment = getLatestDeployment(environmentName);
+
+		if (!comments.equalsIgnoreCase(latestDeployment.getComments())) {
```

Better to have "include" check instead of "equalsIgnoreCase", as there are some cases (like when deployment approval is rejected and deployment gets cancelled ) where OneOps will append some specific texts to the user provided comments.

---

Done.